alf/layers.py

```diff
@@ -20,13 +20,22 @@
 import torch
 import torch.nn as nn
 
+import alf
 from alf.initializers import variance_scaling_init
 from alf.nest.utils import get_outer_rank
 from alf.tensor_specs import TensorSpec
 from alf.utils import common
 from alf.utils.math_ops import identity
 
 
+def reduce_along_batch_dims(x, mean, op):
+    spec = TensorSpec.from_tensor(mean)
```

Comment:
from_tensor by default will only treat dim=0 as the batch dim. Then there is no need to use BatchSquash to flatten the tensor x. Generally I feel that this function is not that necessary.

Reply:
mean is actually the tensor that has dim=0 as the batch dim, while x can include batch_size, batch_length two batch dims.

This function is modeled after the normalize_along_batch_dims function below it.

Reply:
I think it's for specific purpose so probably it should be an inner function in whatever calls it and prefixed by '_'.